src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -379,6 +367,24 @@ internal void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Bu
 
             lock (_sync)
             {
+                // If the reader is completed
+                if (_readerCompletion.IsCompleted)
+                {
+                    ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed();
+                }
+
+                var fullBuffer = new ReadOnlySequence<byte>(_readHead, _readHeadIndex, _commitHead, _commitHeadIndex - _commitHead.Start);
+                var examinedToConsumed = fullBuffer.Slice(examined, consumed);
```

Comment:
Can this throw a bad exception if somebody passed in bad cursors (like mixed up examined and consumed).

Reply:
Yes, out of bounds or end cursor not reachable.

Reply:
What if examined < consumed?